genie-web/src/main/java/com/netflix/genie/web/aspect/HealthCheckMetricsAspect.java

```diff
@@ -0,0 +1,248 @@
+            joinPoint.getTarget().getClass().getSimpleName(),
+            turnaround,
+            throwable != null ? throwable.getClass().getSimpleName() : "none"
+        );
+        final Map<String, String> tags;
+        if (throwable == null) {
+             tags = MetricsUtils.newSuccessTagsMap();
+        } else {
+            tags = MetricsUtils.newFailureTagsMapForException(throwable);
+        }
+        tags.put(HEALTH_INDICATOR_CLASS_TAG_NAME, joinPoint.getTarget().getClass().getSimpleName());
+        registry
+            .timer(healthIndicatorTimerId.withTags(tags))
+            .record(turnaround, TimeUnit.NANOSECONDS);
+    }
+
+    /**
+     * Intercept calls to the main AbstractHealthAggregator and publish counters for number of invocation and failures,
+     * tagged with status and indicator name.
+     * @param joinPoint joinPoint for the aggregateDetails() call
+     */
+    @Before(
+        "execution("
+            + "  java.util.Map<String, Object> "
+            + "  org.springframework.boot.actuate.health.AbstractHealthAggregator.aggregateDetails("
+            + "    java.util.Map<String, org.springframework.boot.actuate.health.Health>"
+            + "  )"
+            + ")"
+    )
+    public void abstractHealthAggregatorAggregateDetailsMonitor(final JoinPoint joinPoint) {
+
+        final Map<String, Health> healthDetailsMap;
+        try {
+            @SuppressWarnings("unchecked")
+            final Map<String, Health> map = (Map<String, Health>) joinPoint.getArgs()[0];
+            healthDetailsMap = map;
+        } catch (final Throwable t) {
+            log.warn(
+                "Failed to cast AbstractHealthAggregator health details argument: {}",
+                joinPoint.getArgs(),
+                t
+            );
+            return;
+        }
+
+        healthDetailsMap.forEach(
+            (name, health) -> {
+
+                final HashMap<String, String> tags = Maps.newHashMap();
+                tags.put(HEALTH_INDICATOR_NAME_TAG_NAME, name);
+                tags.put(MetricsConstants.TagKeys.STATUS, health.getStatus().getCode());
+                final boolean isUp = Status.UP.equals(health.getStatus());
+
+                // Count individual health-check executed, and publish tagged with name and status.
+                // Good for dashboards, can be grouped and filtered by either.
+                registry
+                    .counter(healthIndicatorCounterId.withTags(tags))
+                    .increment();
+
+                //Count failures (if a healthcheck passees, this counter is incremented by 0), publish tagged with
+                // name and status.
+                // Good for alerting, any aggregate that is greater than zero for some period of time signals a
+                // check that is consistently failing.
+                registry
+                    .counter(healthIndicatorFailureCounterId.withTags(tags))
+                    .increment(isUp ? 0 : 1);
```

Comment:
Increment 0 just makes it so that the value doesn't increase. So are you saying that your rate becomes 0 not your absolute value?

Reply:
This metric counts the number of failed healtchecks, I publish zeroes so that the metric is present even if there are no failures, making it easy to see that queries/graphs/alerts are working.